x/smart/stream_dialer.go

```diff
@@ -253,18 +255,50 @@ func (f *StrategyFinder) testDialer(ctx context.Context, dialer transport.Stream
 		testCtx, cancel := context.WithTimeout(ctx, f.TestTimeout)
 		defer cancel()
 
+		// Dial
+
 		testConn, err := dialer.DialStream(testCtx, testAddr)
 		if err != nil {
 			f.logCtx(ctx, "🏁 failed to dial: '%v' (domain: %v), duration=%v, dial_error=%v ❌\n", transportCfg, testDomain, time.Since(startTime), err)
 			return err
 		}
+
+		// TLS Connection
```

Comment:
We don't need the TLS test if we are doing the HTTP fetch.

Reply:
Ah actually we're using this `tlsConn` to make the HTTPS request below. Using just `testConn` without the tunnel setup fails.

So we may as well keep the separate error logging here for better insight into where the connection is failing.